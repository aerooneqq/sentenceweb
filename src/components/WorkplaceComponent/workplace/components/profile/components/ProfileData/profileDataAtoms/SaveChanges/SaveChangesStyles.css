#saveChangesBtn{ 
    width: 150px;
    height: 40px;
    background-color: white;

    font-size: 19px;
    color: orange;

    border-style: solid;
    border-radius: 5px;
    border-width: 1px;
    border-color: orange;

    margin-left: 5%;

    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

    #saveChangesBtn:hover{ 
        border-color: #F97646;
    }

    #saveChangesBtn:hover #saveChangesBtnIcon{ 
        background: url("../../../../img/saveChangesActive.png")
    }

    #saveChangesBtn:hover #saveChangesBtnText{ 
        color: #F97646;
    }

    #saveChangesBtn:focus{ 
        outline: none;
    }

#saveChangesBtnIcon{ 
    background: url("../../../../img/saveChangesStatic.png");
    width: 23px; 
    height: 23px;
}

#saveChangesBtnText{ 
    font-family: Arial; 
    margin-left: 5px;
    font-size: 17px;
    color: orange;
}