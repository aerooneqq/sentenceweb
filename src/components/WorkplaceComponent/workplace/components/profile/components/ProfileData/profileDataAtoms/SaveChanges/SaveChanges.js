import React, {Component} from "react";

//Styles
import "./SaveChangesStyles.css";

export default class SaveChanges extends Component{ 
    constructor(props){ 
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() { 
        this.props.saveChanges();
    }

    render() { 
        return(
            <button id = "saveChangesBtn" onClick={this.handleClick}>
                <div id = "saveChangesBtnIcon"></div>    
                <div id = "saveChangesBtnText">Save</div>
            </button>
        );
    }
}