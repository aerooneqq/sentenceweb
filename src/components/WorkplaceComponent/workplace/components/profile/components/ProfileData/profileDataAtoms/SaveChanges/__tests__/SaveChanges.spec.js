import React from "react";
import {shallow} from "enzyme"; 
import {shallowToJson} from "enzyme-to-json";
import SaveChanges from "../SaveChanges";

const spy = jest.fn(() => {});
const props = { 
    saveChanges: spy
}

describe("SaveChanges", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<SaveChanges />))).toMatchSnapshot();
    });

    it("should call saveChanges when clicked", () => { 
        const saveChanges = shallow(<SaveChanges {...props} />);

        saveChanges.find("button").simulate("click");

        expect(spy).toBeCalled();
    });
});