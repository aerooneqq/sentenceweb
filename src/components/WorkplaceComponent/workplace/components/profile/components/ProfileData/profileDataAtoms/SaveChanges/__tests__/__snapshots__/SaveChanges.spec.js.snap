// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveChanges should render correctly 1`] = `
<button
  id="saveChangesBtn"
  onClick={[Function]}
>
  <div
    id="saveChangesBtnIcon"
  />
  <div
    id="saveChangesBtnText"
  >
    Save
  </div>
</button>
`;
